### PR TITLE
Issue 913 - Save eventlog before deepClean for anax

### PR DESCRIPTION
**Pull request by sirotsinskuy**
Signed-off-by: Antonii Sirotsinskyi <asiro@softserveinc.com>

### CODE REVIEW

File: cli/unregister/unregister.go [hunk 1]
```diff
@@ -288,3 +302,65 @@ func RemoveServiceContainers() error {
 		return fmt.Errorf(err_string)
 	}
 }
+
+// backupEventLogs loads eventlogs from anax.db, marshals them into the JSON format
+// and saves the bkp file into horizon folder with name of backup time
+func backupEventLogs() error {
```
Comment (linggao) [inline cli/unregister/unregister.go:308]: Please call anax /eventlog REST api to get the eventlog and save the output to a text file.  Check this function to see how it is done. Check the case where all=true, detail=true, selections={}, tailing=false.
https://github.com/open-horizon/anax/blob/master/cli/eventlog/eventlog.go#L66.

Comment (linggao) [inline cli/unregister/unregister.go:308]: Please call anax REST API /eventlog to get the event logs and then save it to a text file.  Check this function to see how to get event logs from anax. https://github.com/open-horizon/anax/blob/master/cli/eventlog/eventlog.go#L66

File: cli/unregister/unregister.go [hunk 2]
```diff
@@ -288,3 +298,38 @@ func RemoveServiceContainers() error {
...
+	if err != nil {
+		return fmt.Errorf("failed to backup eventlogs file %v. %v", fileName, err)
+	}
+	defer file.Close()
+
+	if _, err := file.Write(elogsJson); err != nil {
+		return fmt.Errorf("failed to save the eventlogs to file %v. %v", fileName, err)
+	}
+
+	_ = os.Remove(path.Join(hznDir, "anax_tmp.db"))
```
[the remark this answers is not in the document]
Comment (sirotsinskuy) [reply cli/unregister/unregister.go]: That was related to previous approach. Has been removed now

**Comment by sirotsinskuy** [comment]
@linggao PR has been updated. Please look into when have a time.

File: cli/unregister/unregister.go [hunk 3]
```diff
@@ -34,6 +38,13 @@ func DoIt(forceUnregister, removeNodeUnregister bool, deepClean bool, timeout in
 	horDevice := api.HorizonDevice{}
 	cliutils.HorizonGet("node", []int{200}, &horDevice, false)
 
+	if deepClean {
+		err := backupEventLogs()
+		if err != nil {
+			msgPrinter.Printf("Cannot backup eventlogs: %v", err)
```
Comment (linggao) [inline cli/unregister/unregister.go:44]: Please add  `msgPrinter.Println()` here to print out a newline.

Comment (sirotsinskuy) [reply cli/unregister/unregister.go:44]: Added printing of new line for both of `Printf` usages

File: cli/unregister/unregister.go [hunk 4]
```diff
@@ -288,3 +300,39 @@ func RemoveServiceContainers() error {
 		return fmt.Errorf(err_string)
 	}
 }
+
+// backupEventLogs loads eventlogs from eventlog API , marshals them into the JSON format
+// and saves the bkp file into horizon folder with name of backup time
+func backupEventLogs() error {
+	hznDir := "/var/horizon/"
+
```
Comment (linggao) [inline cli/unregister/unregister.go:308]: On Mac where anax is running inside a docker container and hzn is running on the host, this directory does not exist. And you may run into the permission problem for different platforms. Please change it to "/tmp/" to be safe.

Comment (sirotsinskuy) [reply cli/unregister/unregister.go]: Updated directory to `/tmp/`

**Pull request merged**